core/src/apps/nem2/namespace/layout.py

```diff
@@ -97,35 +97,43 @@
     if not embedded:
         await require_confirm_final(ctx, common.max_fee)
 
-async def require_confirm_properties(ctx, creation: NEM2MosaicAliasTransaction):
+async def ask_mosaic_alias(
+    ctx,
+    common: NEM2TransactionCommon,
+    mosaic_alias: NEM2NamespaceRegistrationTransaction,
+    embedded=False
+):
     properties = []
 
     # Mosaic ID
-    if creation.mosaic_id:
+    if mosaic_alias.mosaic_id:
         t = Text("Confirm properties", ui.ICON_SEND, new_lines=False)
         t.bold("Mosaic Id:")
         t.br()
-        t.normal(creation.mosaic_id)
+        t.normal(mosaic_alias.mosaic_id)
         properties.append(t)
 
     # Namespace ID
-    if creation.namespace_id:
+    if mosaic_alias.namespace_id:
         t = Text("Confirm properties", ui.ICON_SEND, new_lines=False)
         t.bold("Namespace Id:")
         t.br()
-        t.normal(creation.namespace_id)
+        t.normal(mosaic_alias.namespace_id)
         properties.append(t)
     # Alias Action
-    if creation.alias_action:
-        if creation.alias_action:
+    if mosaic_alias.alias_action:
+        if mosaic_alias.alias_action:
             alias_text = "Link"
         else:
             alias_text = "Unlink"
         t = Text("Confirm properties", ui.ICON_SEND, new_lines=False)
         t.bold("Alias Action:")
         t.br()
-        t.normal('{} ({})'.format(alias_text, creation.alias_action))
+        t.normal('{} ({})'.format(alias_text, mosaic_alias.alias_action))
         properties.append(t)
 
     paginated = Paginated(properties)
-    await require_confirm(ctx, paginated, ButtonRequestType.ConfirmOutput)
\ No newline at end of file
+    await require_confirm(ctx, paginated, ButtonRequestType.ConfirmOutput)
+
+    if not embedded:
+        await require_confirm_final(ctx, common.max_fee)
```

Comment:
I think I fixed this stuff up in my branch too so hopefully it just auto merges nicely